tensorrt_llm/_torch/models/checkpoints/mx/checkpoint_loader.py

```
@@ -0,0 +1,374 @@
+        """Standard HF disk loading fallback."""
+        if reason is not None:
+            logger.info(
+                "MX P2P unavailable (%s); loading from disk: %s",
+                reason,
+                checkpoint_dir,
+            )
+        else:
+            logger.info("MX P2P unavailable; loading from disk: %s", checkpoint_dir)
+        return super().load_weights(checkpoint_dir, mapping=mapping, **kwargs)
+
+    def publish_as_source(self, model, mapping: Mapping = None, checkpoint_dir: str = None) -> None:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)